src/tiled/templategroupdocument.cpp

```diff
@@ -94,3 +101,124 @@ void TemplateGroupDocument::addTemplate(ObjectTemplate *objectTemplate)
 {
     mTemplateGroup->addTemplate(objectTemplate);
 }
+
+static void writeTemplateDocumentsXml(QFileDevice *device,
+                                      const TemplateDocuments &templateDocuments)
+{
+    QXmlStreamWriter writer(device);
+
+    writer.setAutoFormatting(true);
+    writer.setAutoFormattingIndent(1);
+
+    writer.writeStartDocument();
+    writer.writeStartElement(QLatin1String("templatedocuments"));
+
+    for (const TemplateGroupDocument *templateDocument : templateDocuments) {
+        writer.writeStartElement(QLatin1String("templatedocument"));
+
+        writer.writeAttribute(QLatin1String("path"), templateDocument->fileName());
```

Comment:
Please use a relative path here (can check the `MapWriter` for examples regarding external tilesets and tileset images).